### PR TITLE
first successful test implementation

### PR DESCRIPTION
**Pull request by gigicupp**
@masoncpott Hi Mason. Sorry for bunch of code changes. Going forward, now we get how the PR is supposed work, I promise to make smaller PRs. Can you please review what I have so far?
In this PR:
1. I sat up the Jest testing on the server
2. I updated my endpoint to have the productID randomly generated.
Thank you!

### CODE REVIEW

File: client/dist/index.html [hunk 1]
```diff
@@ -4,6 +4,7 @@
 <head>
     <meta charset="UTF-8">
     <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <link rel="stylesheet" href="app.css">
```
Comment (masoncpott) [inline client/dist/index.html:7]: @gigicupp So this change and also    the css file that you pushed up are not related to testing (the title of this pull request). I know we are all still getting the hang of pull requests and how they are supposed to work, so in the future, try to separate these two file changes into a pull request that solely deals with css/front end work.
BUT, that being said, this and your css look find, it looks like you are just doing some basic experimentation to get some design and structure set up so I don't want to be too critical of it yet. Good job on all your progress!

File: client/src/components/locations.jsx [hunk 2]
```diff
@@ -1,15 +1,46 @@
...
+        <br/>
+        {display}
+      </div>
     )
   }
 }
 
-export default Locations
\ No newline at end of file
+export default Locations;
```
Comment (masoncpott) [inline client/src/components/locations.jsx:46]: @gigicupp again,  this should be in a separate pull request that deals only with jsx components or front end material. That being said, great progress! Can't wait to  see more!

File: jestconfig.json [hunk 3]
```diff
@@ -0,0 +1,4 @@
+{
+  "collectCoverage": true,
+  "testEnvironment": "node"
```
Comment (masoncpott) [inline jestconfig.json:3]: @gigicupp this looks good. I am wondering if the "collectCoverage" key is what generates all these coverage reports and makes every test generate so many files. This is something we should both research because the coverage report files are not necessary to keep in our repo, at least I don't think they are necessary.

File: server/index.test.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+const app = require('./index.js');
+
+test('get locations api test', async () => {
+  await request(app)
+    .get('/api/locations?product_id=1')
+    .expect(200)
+    .then(response => {
+      expect(typeof response).toBe('object');
+    });  
+});
```
Comment (masoncpott) [inline server/index.test.js:18]: @gigicupp This looks good! Also nice change to use a querystring param!

I think it would be good practice to add more tests for this going forward, although this is a great start to getting set up. I think you should test that you are indeed getting the correct product_id back from the database, maybe test what happens if you ask for a product_id that doesn't exist, maybe also test that the structure of the data you get back is what you want it to be.

Great start to testing!

File: server/index.js [hunk 5]
```diff
@@ -1,14 +1,24 @@
...
+
 let app = express();
 
 app.use(express.static(__dirname + '/../client/dist'));
 app.use(express.json());
 app.use(express.urlencoded({ extended: true }));
 
-
+app.get('/api/locations', (req, res) => {
+  let productID = req.query.product_id;
```
Comment (masoncpott) [inline server/index.js:11]: @gigicupp could this be done using fancy es6 destructuring?

**Review by masoncpott — CHANGES_REQUESTED** [review]
@gigicupp Great start to testing! Just a few small changes!

File: client/dist/index.html (hunk 1 above)
Comment (gigicupp) [reply client/dist/index.html:7]: @masoncpott will do.

File: client/src/components/locations.jsx (hunk 2 above)
Comment (gigicupp) [reply client/src/components/locations.jsx:46]: @masoncpott yes, definitely will be doing smaller PRs.

File: jestconfig.json (hunk 3 above)
Comment (gigicupp) [reply jestconfig.json:3]: @masoncpott you are absolutely right on this. I went to the official Jest docs and it says having "true" creating those files. So changed it to false.

File: server/index.test.js (hunk 4 above)
Comment (gigicupp) [reply server/index.test.js:18]: @masoncpott added res.sendStatus(404) on the server. So if a user requests a product that doesn't exist, then they will be getting 404 error, instead of just "me" seeing the console.log error. added this test.

File: server/index.js (hunk 5 above)
Comment (gigicupp) [reply server/index.js:11]: @masoncpott lol. next time. on the req.query, I am not really trying to extract anything else.

**Review by masoncpott — APPROVED** [review]
